app/Http/Middleware/Share.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\Models\Setting;
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\View;
+
+class Share
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $data = [];
+        $settings = Setting::all();
+        $settingArr = [];
+        foreach ($settings as $setting) {
+            $settingArr[$setting->key] = $setting->value;
+        }
+        $data["settings"] = $settingArr;
+        View::share($data);
```

Comment:
view share middleware seviyesinde yapılmamalı. middleware'ı amacı dışında kullanmak oluyor. doğru şekilde kullanmak için https://laravel.com/docs/9.x/views#view-composers